records/generator.go

```diff
@@ -241,12 +244,21 @@ func (rg *RecordGenerator) loadWrap(ip string, port string) (state.State, error)
 	if err != nil {
 		return state.State{}, err
 	}
-	if rip := leaderIP(sj.Leader); rip != ip {
-		logging.VeryVerbose.Println("Warning: master changed to " + ip)
-		sj, err = rg.loadFromMaster(rip, port)
-		return sj, err
+	if sj.Leader != "" {
+		// FIXME(prozlach): Still, we are assuming that sj.Leader is a correct
```

Comment:
The format did not change, but if it changes due to i.e. some error condition, this will Bork again.

My point was that it would be nice to have state.json validated before we start processing it so that assumptions like that are caught early and can be quickly reported to Mesos team in case when they become invalid.